libs/designer-ui/src/lib/arrayeditor/plugins/CollapsedArrayValidation.tsx

```diff
@@ -26,17 +25,17 @@ export const CollapsedArrayValidation = ({
   defaultErrorMessage,
   collapsedValue,
   itemSchema,
+  isComplex,
   setIsValid,
   setItems,
   setCollapsedValue,
 }: CollapsedArrayValidationProps): JSX.Element => {
   const [editor] = useLexicalComposerContext();
-  const [errorMessage, setErrorMessage] = useState('');
 
   const onChange = (editorState: EditorState) => {
     editorState.read(() => {
-      if (itemSchema) {
-        serializeComplexArray(editor, itemSchema, setItems as (complexItems: ComplexArrayItems[]) => void, setIsValid, setErrorMessage);
```

Comment:
So no more error message needed?

Reply:
This was a "feature" I had added with the previous complex array editor to identify where the issue was; however, it became more difficult to support passing in the whole itemSchema. Though I think we can add this as a potential feature in the future. Right now it tries to convert the array of complex Items accordingly to the item schema. If it succeeds, we can view the expanded array. If not, the button to switch is blocked and it'll just show the error "Please enter a valid array"